client/components/app-promo/test/index.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+
+import useFakeDom from 'test/helpers/use-fake-dom';
+
+describe( 'AppPromo', ( ) => {
+	useFakeDom();
+
+	const appPromoDetails = {
+		promoCode: 'a0001',
+		message: 'WordPress.com your way  — desktop app now available for Mac, Windows, and Linux.'
+	};
+	const appPromoLink = 'http://www.wordpress.com';
+
+	let AppPromo;
+	let AppPromoComponent;
+	const defaultPropsToRender = {
+		location: 'reader',
+		promoItem: appPromoDetails,
+		getPromoLink: () => appPromoLink
+	};
+	before( ( ) => {
+		AppPromo = require( '..' ).AppPromo;
+		AppPromoComponent = ( <AppPromo { ...defaultPropsToRender } /> );
+	} );
+
+	it( 'should render the primary components', ( ) => {
+		const wrapper = shallow( AppPromoComponent );
+
+		expect( wrapper.find( '.app-promo' ) ).to.have.lengthOf( 1 );
```

Comment:
I think this is fine, but `chai-enzyme` does include some additional assertions to make the testing intent clearer:

```js
expect( wrapper ).to.have.descendants( '.app-promo' );
```

https://github.com/producthunt/chai-enzyme#descendantsselector